CMakeLists.txt

```diff
@@ -44,50 +45,110 @@ INCLUDE_DIRECTORIES(./third_party/libev)
 ADD_LIBRARY(ev STATIC third_party/libev/ev.c)
 TARGET_COMPILE_DEFINITIONS(ev PRIVATE EV_STANDALONE=1)
 TARGET_COMPILE_OPTIONS(ev PRIVATE -w)
-ADD_EXECUTABLE(MempoolUnitTest.test src/Utils/Mempool.hpp test/MempoolUnitTest.cpp)
-ADD_EXECUTABLE(CStrUnit.test src/Utils/CStr.hpp test/CStrUnitTest.cpp)
-ADD_EXECUTABLE(CommonUnit.test src/Utils/Common.hpp test/CommonUnitTest.cpp)
-ADD_EXECUTABLE(TraitsUnit.test src/Utils/Traits.hpp test/TraitsUnitTest.cpp)
-ADD_EXECUTABLE(RefVectorUnit.test src/Utils/RefVector.hpp test/RefVectorUnitTest.cpp)
-ADD_EXECUTABLE(ItrRangeUnit.test src/Utils/ItrRange.hpp test/ItrRangeUnitTest.cpp)
-ADD_EXECUTABLE(Base64Unit.test src/Utils/Base64.hpp test/Base64UnitTest.cpp)
-ADD_EXECUTABLE(BufferUnit.test src/Buffer/Buffer.hpp test/BufferUnitTest.cpp)
-ADD_EXECUTABLE(BufferPerf.test src/Buffer/Buffer.hpp test/BufferPerfTest.cpp)
-ADD_EXECUTABLE(RingUnit.test src/Utils/Ring.hpp test/RingUnitTest.cpp)
-ADD_EXECUTABLE(ListUnit.test src/Utils/List.hpp test/ListUnitTest.cpp)
-ADD_EXECUTABLE(RulesUnit.test src/mpp/Rules.hpp test/RulesUnitTest.cpp)
-ADD_EXECUTABLE(EncDecUnit.test src/mpp/mpp.hpp test/EncDecTest.cpp)
-ADD_EXECUTABLE(Client.test src/Client/Connector.hpp test/ClientTest.cpp)
-ADD_EXECUTABLE(ClientPerfTest.test src/Client/Connector.hpp test/ClientPerfTest.cpp)
-ADD_EXECUTABLE(SimpleExample examples/Simple.cpp)
-TARGET_LINK_LIBRARIES(SimpleExample ${COMMON_LIB})
-TARGET_LINK_LIBRARIES(ClientPerfTest.test ${COMMON_LIB})
-TARGET_LINK_LIBRARIES(Client.test ${COMMON_LIB})
+
+OPTION(TNTCXX_BUILD_TESTING
+    "If ON, tntcxx will build all of tntcxx's own tests." OFF)
```

Comment:
AFAIR we planned to make this option `ON'` if tntcxx is top project, and `OFF` otherwise. Can we do so?

Reply:
Fixed, as we discussed. Added an internal flag `TNTCXX_BUILD_TESTING_DEFAULT`, which is set to true, if the project is determined to be top level using [`PORJECT_IS_TOP_LEVEL`](https://cmake.org/cmake/help/latest/variable/PROJECT_IS_TOP_LEVEL.html).